flink-ml-lib/src/main/java/org/apache/flink/ml/common/param/HasBatchSize.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.common.param;
+
+import org.apache.flink.ml.param.IntParam;
+import org.apache.flink.ml.param.Param;
+import org.apache.flink.ml.param.ParamValidators;
+import org.apache.flink.ml.param.WithParams;
+
+/** Interface for the shared batchSize param. */
+public interface HasBatchSize<T> extends WithParams<T> {
+
+    Param<Integer> BATCH_SIZE =
+            new IntParam(
+                    "batchSize", "Batch size of training algorithms.", 100, ParamValidators.gt(0));
```

Comment:
Hi Dong, thanks for the comments. The doc you mentioned above is talking about setting batch size for a single machine with multiple cores. (Since the number of cores is usually power of 2.)

However, doing machine learning on big data engines like Spark/Flink is a bit different from doing machine learning on a single machine --- We would have arbitary number of parallel instances.

I have no preference on the default value of batch size. If you prefer using 32 as the default value, please let me know.

Reply:
Hmm.. I think even if we run distributed machine learning algorithm on multiple workers, each worker will still process one batch of records at a time, right? If so, we probably still want the batch size to be a multiple of the number of CPU cores on a single machine.

And yes, I think batchSize=32 is probably a good default value, based on the discussion in the StackExchange link. Thanks!

Reply:
Hmm... in addition to renaming this class, could the parameter name also be renamed to `globalBatchSize`?